src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,102 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double MULTIPLE_CONSTANT = 1.5;
+    private int size;
+    private T[] array;
+
+    public ArrayList() {
+        array = (T[]) new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
+
+    public void grow() {
+        T[] newArray = (T[]) new Object[(int) (array.length * MULTIPLE_CONSTANT)];
+        System.arraycopy(array, 0, newArray, 0, array.length);
+        array = newArray;
+    }
 
     @Override
     public void add(T value) {
-
+        if (size == array.length) {
+            grow();
+        }
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayIndexOutOfBoundsException();
+        }
+        if (index == size) {
+            add(value);
+        } else {
+            System.arraycopy(array, index, array, index + 1, size - index);
+            array[index] = value;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index < 0 || index >= size) {
+            throw new ArrayIndexOutOfBoundsException();
+        }
+        return array[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (index < 0 || index >= size) {
+            throw new ArrayIndexOutOfBoundsException();
+        }
+        array[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        if (index < 0 || index >= size) {
+            throw new ArrayIndexOutOfBoundsException("Can't remove element with index: " + index);
+        }
```

Comment:
we can move this to separate method(you are using the same check in three places